presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -210,32 +211,27 @@ private InternalTable buildTables(Session session, String catalogName, Map<Strin
 
     private InternalTable buildTablePrivileges(Session session, String catalogName, Map<String, NullableValue> filters)
     {
-        List<GrantInfo> grants = ImmutableList.copyOf(getTablePrivilegesList(session, catalogName, filters));
+        QualifiedTablePrefix prefix = extractQualifiedTablePrefix(catalogName, filters);
+        String grantee = extractGrantee(session, filters);
+        List<GrantInfo> grants = ImmutableList.copyOf(listTablePrivileges(session, metadata, accessControl, prefix, grantee));
 
         InternalTable.Builder table = InternalTable.builder(informationSchemaTableColumns(TABLE_TABLE_PRIVILEGES));
         for (GrantInfo grant : grants) {
-            if (session.getIdentity().getUser().equals(grant.getIdentity().getUser())) {
```

Comment:
You probably want to keep it just in this method, you don't want this check in as general a place as MetadataListing